### ISSUE TITLE
Istio 1.19 sends a duplicate prometheus metric and may cause datadog scrape to fail

### ISSUE THREAD

**Issue by tehlers320** [istio/istio#46977]
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

```
# HELP istio_build Istio component build info
# TYPE istio_build gauge
istio_build 0
istio_build{component="pilot",tag="1.19.0"} 1
```

This causes agents such as datadog to crash their scrape because we have a nil.
https://github.com/DataDog/integrations-core/blob/master/istio/datadog_checks/istio/metrics.py#L383C1-L383C15

```
    scraper.scrape()\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/openmetrics/v2/scraper.py\", line 254, in scrape\n    transformer(metric, self.generate_sample_data(metric), runtime_data)\n  File \"/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/openmetrics/v2/transformers/metadata.py\", line 24, in metadata\n    set_metadata_method(metric_name, sample.labels[label], **options)\nKeyError: 'tag'\n"}]
```

1.17 vs 1.19 metrics 
```
diff <(cat /tmp/1.19  |cut -d \{ -f1 |sort |uniq | awk '{print $1}') <(cat /tmp/1.17  |cut -d \{ -f1 |awk '{print  $1}' |sort |uniq)  |grep -v '#'
2,154c2,5
---
> callback_time
> callback_time_duration_bucket
> callback_time_duration_count
> callback_time_duration_sum
155a7
> citadel_server_csr_count
156a9
> citadel_server_success_cert_issuance_count
186a40,46
> grpc_server_handled_total
> grpc_server_handling_seconds_bucket
> grpc_server_handling_seconds_count
> grpc_server_handling_seconds_sum
> grpc_server_msg_received_total
> grpc_server_msg_sent_total
> grpc_server_started_total
188d47
< istio_build
190d48
< istiod_managed_clusters
192a51,52
> pilot_conflict_outbound_listener_http_over_current_tcp
> pilot_conflict_outbound_listener_tcp_over_current_http
198d57
< pilot_dns_cluster_without_endpoints
204c63
< pilot_info
---
> pilot_jwks_resolver_network_fetch_success_total
211a71,73
> pilot_proxy_endpointslice_time_bucket
> pilot_proxy_endpointslice_time_count
> pilot_proxy_endpointslice_time_sum
218a81
> pilot_sds_certificate_errors_total
222,229d84
< pilot_worker_queue_depth
< pilot_worker_queue_depth
< pilot_worker_queue_duration_bucket
< pilot_worker_queue_duration_count
< pilot_worker_queue_duration_sum
< pilot_worker_queue_latency_bucket
< pilot_worker_queue_latency_count
< pilot_worker_queue_latency_sum
231,232d85
< pilot_xds
< pilot_xds_cds_reject
236,237c89
< pilot_xds_eds_reject
< pilot_xds_lds_reject
---
> pilot_xds_expired_nonce
242d93
< pilot_xds_rds_reject
253,254c104,106
< startup_duration_seconds
< wasm_cache_entries
---
> sidecar_injection_requests_total
> sidecar_injection_skip_total
> sidecar_injection_success_total
256c108
< webhook_patch_failures_total
```

Prometheus works fine and ideally this is a datadog problem. But it would be nice to not send duplicate metrics.

### Version

```Text
1.19.0 only
```

### Additional Information

@howardjohn is already looking at this and has a potential fix OTW. 

Uesrs of datadog who must have 1.19 may be able to simply exclude `istio_build` and `pilot_info` here: https://github.com/DataDog/integrations-core/blob/master/istio/datadog_checks/istio/data/conf.yaml.example#L133 (i have not tested this, but it looks like it will fix it).

**Comment by howardjohn** [comment]
yeah I have a fix almost ready

**Comment by JacksonDavenport** [comment]
Hey thanks for the quick fix on this. Just chiming in from the Datadog side that yes this will impact the metric collection for the control plane (istiod) portion of the Istio check. This should not impact the data plane portion for reporting all the `istio.mesh.*` metrics.

Confirming as well that workaround. You can update your configuration to exclude this metric for now. For example:

```yaml
ad.datadoghq.com/discovery.check_names: '["istio"]'
ad.datadoghq.com/discovery.init_configs: '[{}]'
ad.datadoghq.com/discovery.instances: |
  [
    {
      "istiod_endpoint": "http://%%host%%:15014/metrics",
      "use_openmetrics": "true", 
      "exclude_metrics": ["istio_build"]
    }
  ] 
```

**Comment by howardjohn** [comment]
This will be fixed in 1.19.1, and was introduced in 1.19.0. So only 1.19.0 users should be impacted. Thanks for all the collaboration on this, everyone.
